rootfs/etc/haproxy/template/haproxy.tmpl

```diff
@@ -108,6 +108,18 @@ frontend httpfront
 {{ range $server := $ing.HAServers }}
 {{ if $server.UseHTTP }}
 {{ range $location := $server.Locations }}
+    http-request set-var(req.ingress_orig_path) path
+    acl {{ $location.Backend }}_path_match var(req.ingress_orig_path) -m beg {{ $location.Path }} if or ishost-{{ $server.HostnameLabel }} ishost-{{ $server.HostnameLabel }}-alias
+{{ $rewriteTarget := $location.Rewrite.Target }}
+{{ if ne $rewriteTarget "" }}
+{{ if ne $server.Alias "" }}
+    reqrep ^([^\ :]*)\ {{ $location.Path }}[/]?(.*) \1\ {{ $rewriteTarget }}\2 if {{ $location.Backend }}_path_match ishost-{{ $server.HostnameLabel }} or ishost-{{ $server.HostnameLabel }}-alias
+{{ else }}
+    reqrep ^([^\ :]*)\ {{ $location.Path }}[/]?(.*) \1\ {{ $rewriteTarget }}\2 if {{ $location.Backend }}_path_match ishost-{{ $server.HostnameLabel }}
```

Comment:
These lines are almost the same, use an inline `if` instead.

Prefer http-request if possible, otherwise declare them after all http-requests in order to avoid warning from haproxy cfg parser.

Reply:
Unfortunately, the documentation indicates `http-request` does not support capture groups, hence we had to fall back to the `reqrep` option instead.